src/test/java/com/example/petbuddybackend/utils/HeaderUtilsTest.java

```diff
@@ -0,0 +1,118 @@
+
+    @Test
+    void testGetHeaderSingleValue_invalidType_shouldThrowInvalidWebSocketHeaderException() {
+        Map<String, Object> headers = createHeadersWithSingleValue("testHeader", "testValue");
+        assertThrows(InvalidWebSocketHeaderException.class, () ->
+                HeaderUtils.getHeaderSingleValue(headers, "testHeader", Integer.class));
+    }
+
+    @Test
+    void testGetHeaderSingleValue_emptyHeader_shouldThrowInvalidWebSocketHeaderException() {
+        Map<String, Object> headers = createHeadersWithEmptyList("testHeader");
+        assertThrows(InvalidWebSocketHeaderException.class, () ->
+                HeaderUtils.getHeaderSingleValue(headers, "testHeader", String.class));
+    }
+
+    @Test
+    void testGetHeaderSingleValue_multipleValuesInHeader_shouldThrowInvalidWebSocketHeaderException() {
+        Map<String, Object> headers = createHeadersWithMultipleValues("testHeader", List.of("value1", "value2"));
+        assertThrows(InvalidWebSocketHeaderException.class, () ->
+                HeaderUtils.getHeaderSingleValue(headers, "testHeader", String.class));
+    }
+
+    @Test
+    void testGetHeaderSingleValue_invalidHeaderStructure_shouldThrowIllegalArgumentException() {
+        Map<String, Object> headers = Collections.emptyMap();
+        assertThrows(IllegalArgumentException.class, () ->
+                HeaderUtils.getHeaderSingleValue(headers, "testHeader", String.class)
+        );
+
+    }
+
+    @Test
+    void testGetOptionalHeaderSingleValue_validHeader_shouldSucceed() {
+        Map<String, Object> headers = createHeadersWithSingleValue("testHeader", "testValue");
+        Optional<String> result = HeaderUtils.getOptionalHeaderSingleValue(headers, "testHeader", String.class);
+        assertTrue(result.isPresent());
+        assertEquals("testValue", result.get());
+    }
+
+    @Test
+    void testGetOptionalHeaderSingleValue_missingHeader_shouldReturnEmptyOptional() {
+        Map<String, Object> headers = createHeadersWithSingleValue("anotherHeader", "testValue");
+        Optional<String> result = HeaderUtils.getOptionalHeaderSingleValue(headers, "testHeader", String.class);
+        assertFalse(result.isPresent());
+    }
+
+    @Test
+    void testGetOptionalHeaderSingleValue_invalidType_shouldThrowInvalidWebSocketHeaderException() {
+        Map<String, Object> headers = createHeadersWithSingleValue("testHeader", "testValue");
+        assertThrows(InvalidWebSocketHeaderException.class, () ->
+                HeaderUtils.getOptionalHeaderSingleValue(headers, "testHeader", Integer.class));
+    }
+
+    @Test
+    void testGetHeaderSingleValue_enumPassed_shouldReturnEnum() {
+        Map<String, Object> headers = createHeadersWithSingleValue("Accept-Role", "CLIENT");
+        Role role = HeaderUtils.getHeaderSingleValue(headers, "Accept-Role", Role.class);
+        assertNotNull(role);
+    }
+
+    private Map<String, Object> createHeadersWithSingleValue(String headerName, Object value) {
```

Comment:
wszystkie prywatne do MockUtilsow mozna, chociaz tam i tak sie taki tasiemiec zrobil, zeby nwm czy nie porozdzielac tego wszystkiego na kilka klas MockUtils

Reply:
Dobry pomysł, porozdzielam to jakoś sensownie